### PR TITLE
Migrate Category field to woocommerce/taxonomy-field block

### LINKED ISSUE #40019: Migrate Category field to woocommerce/taxonomy-field block
## Description

At https://github.com/woocommerce/woocommerce/pull/39947 we created a taxonomy-field block that is used for the Brands plugin. It was based on the Categories field, but made to be reusable. Currently the Categories field is not using it. We should migrate it and delete the category field, making sure that no regressions occur.

## Acceptance criteria

- [ ] The categories field is using the woocommerce/taxonomy-field block without any functionality regressions and no major UX changes

### PR DESCRIPTION
**Pull request by nathanss**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Migrate Category field to woocommerce/taxonomy-field block

Closes #40019 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the new product editor
2. Go to "Organization" tab
3. Test the "Categories" field:
4. Select from existing categories
5. Create a new category with and without parent
6. Save the product

Do the same scenario for the trunk branch and this branch and write down if you detect any differences in behavior. 

Known (and acceptable) differences:
* Error messages from "create new" dialog now appear at the bottom of the name field
* The "Parent" field at "create new" now shows hierarchy

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/products/fills/details-section/details-section-fills.tsx [hunk 1]
```diff
@@ -44,13 +43,6 @@ export const DetailsSectionFills = () => (
 		>
 			<DetailsNameField />
 		</WooProductFieldItem>
-		<WooProductFieldItem
```
Comment (nathanss) [inline plugins/woocommerce-admin/client/products/fills/details-section/details-section-fills.tsx:47]: Is it OK to remove this?

Comment (mattsherman) [reply plugins/woocommerce-admin/client/products/fills/details-section/details-section-fills.tsx:47]: I believe so... this is from the old new product editor, right?

Comment (mattsherman) [reply plugins/woocommerce-admin/client/products/fills/details-section/details-section-fills.tsx:47]: We really should get rid of all of that old new product editor code in a separate PR!

File: plugins/woocommerce/src/Admin/Features/ProductBlockEditor/BlockRegistry.php [hunk 2]
```diff
@@ -22,7 +22,6 @@ class BlockRegistry {
 	const PRODUCT_BLOCKS = [
 		'woocommerce/conditional',
 		'woocommerce/product-catalog-visibility-field',
-		'woocommerce/product-category-field',
```
Comment (mattsherman) [inline plugins/woocommerce/src/Admin/Features/ProductBlockEditor/BlockRegistry.php:25]: Should we add an entry for `woocommerce/taxonomy-field`?

I'm not really sure why it works without it, or what any problem would be.

**Review by mattsherman — CHANGES_REQUESTED** [review]
Overall, works largely as expected. Love all the removed code!

Left a comment or two.

In addition, I noticed the following differences:

- The style of the "Save" button in the create modal is incorrect when saving
    - We are missing a `setIsCreating( true )` call in `CreateTaxonomyModal::onSave()`

- If you create a new category and set a parent for it:
    - The old category field selects the parent when you save the new category
    - The new taxonomy field does not select the parent when you save the new category
    - I'm not really sure which one is "correct"... we don't auto-select the parent in any other scenario, so I'm leaning towards saying that the new behavior is actually correct

**Comment by nathanss** [comment]
> * If you create a new category and set a parent for it:
>   
>   * The old category field selects the parent when you save the new category
>   * The new taxonomy field does not select the parent when you save the new category
>   * I'm not really sure which one is "correct"... we don't auto-select the parent in any other scenario, so I'm leaning towards saying that the new behavior is actually correct

@mattsherman I have never noticed that in the old field! I also think the new behavior is correct! I believe I addressed all your other suggestions. Thank you for the review!

**Review by mattsherman — APPROVED** [review]
Nice work on this, @nathanss ! :shipit:

**Comment by rodelgc** [comment]
Hi @nathanss @mattsherman, just one quick question: the changes in this PR only affects the new product block editor, if I understand it correctly? Or does it also impact the categories in the legacy product editor?

The PR touched a lot of files and made changes across them, so just verifying with you both if this would have any impact on the legacy experience. 

For the meantime, I'm adding the `needs: internal testing` label with `impact: high` since the PR contains many changes even though it's behind a feature flag. I'll remove `impact: high` if the changes turn out to be not impacting the legacy experience.

**Comment by mattsherman** [comment]
@rodelgc Correct, this PR only affects the new product block editor. It does not affect the legacy product editor.

**Pull request merged**